src/crystal/system/unix/file.cr

```diff
@@ -70,26 +70,65 @@ module Crystal::System::File
     tmpdir.rchop(::File::SEPARATOR)
   end
 
-  def self.stat?(path : String) : ::File::Stat?
+  def self.stat?(path : String) : ::File::Info?
     if LibC.stat(path.check_no_null_byte, out stat) != 0
       if {Errno::ENOENT, Errno::ENOTDIR}.includes? Errno.value
         return nil
       else
         raise Errno.new("Unable to get stat for '#{path}'")
       end
     end
-    ::File::Stat.new(stat)
+
+    to_file_info(stat)
   end
 
-  def self.lstat?(path : String) : ::File::Stat?
+  def self.lstat?(path : String) : ::File::Info?
     if LibC.lstat(path.check_no_null_byte, out stat) != 0
       if {Errno::ENOENT, Errno::ENOTDIR}.includes? Errno.value
         return nil
       else
         raise Errno.new("Unable to get lstat for '#{path}'")
       end
     end
-    ::File::Stat.new(stat)
+
+    to_file_info(stat)
+  end
+
+  def self.to_file_info(stat)
+    size = stat.st_size.to_u64
+
+    permissions = ::File::Permissions.new((stat.st_mode & 0o777).to_i16)
+
+    case stat.st_mode & LibC::S_IFMT
+    when LibC::S_IFBLK
+      type = ::File::Type::BlockDevice
+    when LibC::S_IFCHR
+      type = ::File::Type::CharacterDevice
+    when LibC::S_IFDIR
+      type = ::File::Type::Directory
+    when LibC::S_IFIFO
+      type = ::File::Type::Pipe
+    when LibC::S_IFLNK
+      type = ::File::Type::SymbolicLink
+    when LibC::S_IFREG
+      type = ::File::Type::File
+    when LibC::S_IFSOCK
+      type = ::File::Type::Socket
+    else
+      raise "BUG: unknown File::Type"
+    end
+
+    flags = ::File::Flags::None
+    flags |= ::File::Flags::SetUser if (stat.st_mode & LibC::S_ISUID) != 0
+    flags |= ::File::Flags::SetGroup if (stat.st_mode & LibC::S_ISGID) != 0
+    flags |= ::File::Flags::Sticky if (stat.st_mode & LibC::S_ISVTX) != 0
+
+    modification_time = ::Time.new(stat.st_mtim, ::Time::Kind::Utc)
```

Comment:
you should better use `::Time.utc`

Reply:
`st_mtim` is a `LibC::Timespec`. You can't use `Time.utc`.

Reply:
ah, nvm